### PR TITLE
HPCC-14420 Dali should initialize cassandra data store

### PR DESCRIPTION
**Pull request by richardkchapman**
Introduce a general mechanism for plugins that partially replace dali
functionality.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith @ghalliday For initial review only - I have not yet finished testing this so do not merge yet

**Comment by richardkchapman** [comment]
@jakesmith Please review. I have tested it and it is working at least in simple cases.

File: plugins/cassandra/cassandrawu.cpp [hunk 1]
```diff
@@ -3030,6 +3030,7 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             cluster.disconnect();
         }
         cacheRetirer.start();
+        LINK(_dll);  // Yes, this leaks. Not really sure how to avoid that.
```
Comment (jakesmith) [inline plugins/cassandra/cassandrawu.cpp]: Can it not be released at ~CCasssandraWorkUnitFactory() time?

**Comment by jakesmith** [comment]
1 Minor question, otherwise looks good.
@richardkchapman

File: plugins/cassandra/cassandrawu.cpp (hunk 1 above)
Comment (richardkchapman) [inline plugins/cassandra/cassandrawu.cpp]: Not really, because that would unload code that was currently executing...

**Comment by richardkchapman** [comment]
@ghalliday Over to you...

File: dali/base/daclient.cpp [hunk 2]
```diff
@@ -231,6 +231,27 @@ void disconnectLogMsgListenerFromDali()
...
+        ForEach(*instances)
+        {
+            IPropertyTree *inst = &instances->query();
+            const char *ps = inst->queryProp("@port");
+            unsigned port = ps?atoi(ps):0;
+            if (!port)
+                port = DALI_SERVER_PORT;
+            SocketEndpoint daliep(inst->queryProp("@netAddress"),port);
+            if (dali.equals(daliep))
+                return cur;;
```
Comment (richardkchapman) [inline dali/base/daclient.cpp]: extra ;

Comment (ghalliday) [inline dali/base/daclient.cpp]: trivial: extra semicolon

**Pull request merged**